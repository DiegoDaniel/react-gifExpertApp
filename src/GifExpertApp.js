import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

  const [categories, setCategories] = useState( ['One Punch'] );

  // const handleAdd = () => {
  //   //Esta forma no da errores, pero me gustÃ³ mas la del profesor.
  //   /*setCategories( categories.concat('Text') );*/
    
  //   //El spread es para 'Extraer TODOS los valores del array'.
  //   /*setCategories( [...categories, 'HunterXHunter'] );*/

  //   //Ademas, el set tiene como callback el ESTADO ANTERIOR: (Esto va a servir)
  //   setCategories( catsPrev => [...catsPrev, 'HunterXHunter'] );

  // }

  return (
      <>
        <h2>GifExpertApp</h2>   
        <AddCategory 
          setCategories={ setCategories }/>
        <hr />
        <ol>
          { categories.map( category => 
              <GifGrid
                category={ category }
                key={ category } />
          )}
        </ol>
      </>
  )
}

export default GifExpertApp;
