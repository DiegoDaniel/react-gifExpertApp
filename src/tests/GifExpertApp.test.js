import React from 'react';
import { shallow } from 'enzyme';

import GifExpertApp from '../GifExpertApp';


describe('Pruebas en <GifExpertApp />', () => {
    

    test('should match snapshot', () => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect( wrapper ).toMatchSnapshot();

    });

    test('should show categories list', () => {
        
        const categories = ['One punch', 'Goku'];
        
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
    
})
