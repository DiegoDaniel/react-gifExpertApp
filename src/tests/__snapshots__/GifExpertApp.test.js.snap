// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> should match snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> should show categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
